tests/WebToPay/CallbackValidatorTest.php

```diff
@@ -60,19 +66,58 @@ public function testValidateAndParseDataWithInvalidProject() {
     }
 
     /**
-     * Tests validateAndParseData method
+     * Tests validateAndParseData method with callback data decoding
      */
-    public function testValidateAndParseData() {
-        $request = array('data' => 'abcdef', 'sign' => 'qwerty');
+    public function testValidateAndParseDataWithDecoding() {
+        $request = array('data' => 'abcdef', 'sign' => 'qwerty', 'ss1' => 'randomChecksum');
         $parsed = array('projectid' => 123, 'someparam' => 'qwerty123', 'type' => 'micro');
 
+        $this->assertArrayHasKey('ss1', $request);
```

Comment:
What a point testing for that? We see `$request` in all its glory above. :)

Reply:
Just wanted to highlight the main criteria based on which flow with encryption would be used. Since tests also perform the function of documentation to some extent. But if you think it is redundant I'm gonna remove that.